docs/usage/exposureclasses.md

```diff
@@ -0,0 +1,147 @@
+  seedSelectors:
+    matchLabels:
+      env: prod
+---
+apiVersion: core.gardener.cloud/v1alpha1
+kind: ExposureClass
+metadata:
+  name: abc
+handler: abc
+scheduling:
+  seedSelector:
+    matchLabels:
+      network: internal
+```
+
+2. Both `seedSelectors` would be merged into the `Shoot`. The result would be the following:
+```yaml
+apiVersion: core.gardener.cloud/v1alpha1
+kind: Shoot
+metadata:
+  name: abc
+  namespace: garden-dev
+spec:
+  exposureClassName: abc
+  seedSelectors:
+    matchLabels:
+      env: prod
+      network: internal
+```
+
+3. Now the Gardener Scheduler would try to find a `Seed` with those labels.
+  - If there are **no** Seeds with matching labels for the seed selector then the `Shoot` will be unschedubale
+  - If there are Seeds with matching labels for the seed selector then the Shoot be assigned to the best candidate after the scheduling strategy is applied, see [here](https://github.com/gardener/gardener/blob/master/docs/concepts/scheduler.md#algorithm-overview)
+    - If the `Seed` is **not** able to serve `ExposureClass` the handler `abc` then the Shoot will end up in error state
+    - If the `Seed` is able to serve the `ExposureClass` handler `abc` then the `Shoot` will be created
+
+</details>
+
+## Gardenlet Configuration `ExposureClass` handlers
+
+The Gardenlet is responsible to realize the control plane exposure strategy defined in the referenced `ExposureClass` of a `Shoot`.
+
+Therefore, the `GardenletConfiguration` can contain an `.exposureClassHandlers` list with the respective configuration.
+
+Example of the `GardenletConfiguration`:
+
+```yaml
+exposureClassHandlers:
+- name: internetconfig
+  loadBalancerService:
+    annotations:
+      loadbalancer/nework: internet
+- name: internalconfig
+  loadBalancerService:
+    annotations:
+      loadbalancer/nework: internal
+  sni:
+    ingress:
+      namespace: ingress-internal
+      labels:
+        network: internal
+```
+
+Each Gardenlet can define how the handler of a certain `ExposureClass` needs to be implemented for the Seed(s) where it is responsible for.
+
+The `.name` is the name of the handler config and it must match to the `.handler` in the `ExposureClass`.
+
+All control planes on a `Seed` are exposed via a load balancer.
+Either a dedicated one or a central shared one.
+The load balancer service needs to be configured in a way that it is reachable from the target network environment.
+Therefore, the configuration of load balancer service need to be configurable which can be done via the `.loadBalancerService` section.
+The common way to influence load balancer service behaviour is via annotations where the respective cloud-controller-manager will react on and configure the infrastucture load balancer accordingly.
+
+In case the Gardenlet runs with activated `APIServerSNI` feature flag (default), the control planes on a `Seed` will be exposed via a central load balancer and with Envoy via TLS SNI pass through.
```

Comment:
```suggestion
In case the Gardenlet runs with activated `APIServerSNI` feature flag (default), the control planes on a `Seed` will be exposed via a central load balancer and with Envoy via TLS SNI passthrough.
```